lib/ae_mdw/aex9.ex

```diff
@@ -123,7 +123,11 @@ defmodule AeMdw.Aex9 do
       contract_id = encode_contract(contract_pk)
 
       paginated_balances =
-        (&Collection.stream(state, Model.Aex9BalanceAccount, &1, key_boundary, cursor_key))
+        fn direction ->
+          state
+          |> Collection.stream(Model.Aex9BalanceAccount, direction, key_boundary, cursor_key)
+          |> Stream.reject(fn {_contract_pk, amount, _account_pk} -> amount == 0 end)
```

Comment:
This is going to cause to potentially have to iterate over an unlimited list of records. To avoid this you can include the minimum of 0 in the key_boundary:
```elixir
      key_boundary = {{contract_pk, 0, <<>>}, {contract_pk, Util.max_int(), <<>>}}
```

Reply:
The amount doesn't go to less than 0, and we have to filter out the ones that equal 0, so with this suggestion, the stream is going the contain the same entries.

Reply:
You're right, amount is an integer, though, right? We can start at 1